agent/config/config.go

```diff
@@ -226,6 +226,11 @@ func NewConfig(ec2client ec2.EC2MetadataClient) (*Config, error) {
 	}
 	config := &envConfig
 
+	if config.OnPrem.Enabled() {
```

Comment:
should we validate that AWSRegion has also been set here so that we don't get into the if-block that relies on the ec2 metadata client below?

Reply:
yes good catch. updated to throw an error if AWSRegion is empty here